src/test/java/bio/terra/service/snapshot/SnapshotTest.java

```diff
@@ -285,4 +298,66 @@ public void snapshotByFullViewHappyPathTest() throws Exception {
         assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
         assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
     }
+
+    @Ignore("We expect this test to be flaky b/c ACLs can take a while to update (up to 7 minutes)")
+    @Test
+    public void snapshotACLTest() throws Exception {
+        DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+
+        String projectId = System.getenv("GOOGLE_CLOUD_DATA_PROJECT");
+        bigQuery = BigQueryOptions.newBuilder()
+            .setProjectId(projectId)
+            .build()
+            .getService();
+
+        // Fetch BQ Dataset
+        String datasetName = dataset.getName();
+        String bqDatasetName = bigQueryPdao.prefixName(datasetName);
+        Dataset bq_dataset = bigQuery.getDataset(bqDatasetName);
+
+        // fetch ACLs
+        List<Acl> beforeAcls = bq_dataset.getAcl();
+        assertEquals("There should be 7 ACLs on the dataset", 7, beforeAcls.size());
```

Comment:
Are the numbers here (7 and 3) not likely to change? If the intent here is to check which ACLs are added and then removed, another approach would be to look at the difference between the ACL lists and verify that the added ACLs are the ones expected. After delete you would verify that the list matches the original list, as captured here. That would avoid a dependence on the initial number of ACLs.

Reply:
Good point! I now compare the difference in ACL count. I think this should make the test more robust.